Documentation/hacking.md

```diff
@@ -96,136 +96,49 @@ Follow the instructions on [Update coreos flavor stage1](devel/update-coreos-sta
-
-### Update godep
-
-Step zero is generally to ensure you have the **latest version** of `godep` available in your `PATH`.
-
-### Having the right directory layout (i.e. `GOPATH`)
-
-To work with `godep`, you'll need to have the repository (i.e. `github.com/coreos/rkt`) checked out in a valid `GOPATH`.
-If you use the [standard Go workflow](https://golang.org/doc/code.html#Organization), with every package in its proper place in a workspace, this should be no problem.
-As an example, if one was obtaining the repository for the first time, one would do the following:
+### Update glide/glide-vc
 
-```
-$ export GOPATH=/tmp/foo               # or any directory you please
-$ go get -d github.com/coreos/rkt/...  # or 'git clone https://github.com/coreos/rkt $GOPATH/src/github.com/coreos/rkt'
-$ cd $GOPATH/src/github.com/coreos/rkt
-```
-
-If, however, you instead prefer to manage your source code in directories like `~/src/rkt`, there's a problem: `godep` doesn't like symbolic links (which is what the rkt build process uses by default to create a self-contained GOPATH).
-Hence, you'll need to work around this with bind mounts, with something like the following:
-
-```
-$ export GOPATH=/tmp/foo        # or any directory you please
-$ mkdir -p $GOPATH/src/github.com/coreos/rkt
-# mount --bind ~/src/rkt $GOPATH/src/github.com/coreos/rkt
-$ cd $GOPATH/src/github.com/coreos/rkt
-```
-
-One benefit of this approach over the single-workspace workflow is that checking out different versions of dependencies in the `GOPATH` (as we are about to do) is guaranteed to not affect any other packages in the `GOPATH`.
-(Using [gvm](https://github.com/moovweb/gvm) or other such tomfoolery to manage `GOPATH`s is an exercise left for the reader.)
-
-### Restoring the current state of dependencies
-
-Now that we have a functional `GOPATH`, use `godep` to restore the full set of vendored dependencies to their correct versions.
-(What this command does is essentially just loop over the set of dependencies codified in `Godeps/Godeps.json`, using `go get` to retrieve and then `git checkout` (or equivalent) to set each to their correct revision.)
-
-```
-$ godep restore # might take a while if it's the first time...
-```
-
-At this stage, your path forks, depending on what exactly you want to do: add, update or remove a dependency, or add, update or remove an application.
-But in _all six cases_, the procedure finishes with the [same save command](#saving-the-set-of-dependencies).
+Ensure you have the **latest version** of `glide` version `>= 0.11` and `glide-vc` available in your `PATH`.
 
 #### Add a new dependency
 
-In this case you'll first need to retrieve the dependency you're working with into `GOPATH`.
-As a simple example, assuming we're adding `github.com/fizz/buzz/tazz`:
-
+Use the glide tool to add a new dependency. In order to add a dependency to a package i.e. `github.com/fizz/buzz` for versions `>=1.2.3` and `<2.0.0`, execute:
```

Comment:
@sdboyer Thanks for the clarification! I guess @lucab wants us to describe here how we want to handle dependency constraints within rkt, which currently is to declare concrete versions (not ranges) in the `glide.yaml` file.

I'll change it accordingly.